Taken care of comments raised by Adrian (#9)
boucadair/policy-based-network-acl: draft-ma-opsawg-ucl-acl.md

      current context, and map the users to their correct access
      entitlement to the network.

>    This document defines a common schedule YANG module in {{sec-schedule}} which

Reviewer: ```suggestion
   This document defines a common schedule YANG module ({{sec-schedule}}) which
```

   is designed to be applicable for policy activation based on date and time
   conditions. This model is designed with the intent to be reusable in other

>    scheduling contexts.

Reviewer: ```suggestion
   scheduling contexts. In particular, the groupings can be reused by other modules.
```

Reviewer: Please fix the pyang issue (see the report at https://github.com/boucadair/policy-based-network-acl/actions/runs/5077408435/jobs/9120637480)

Author: Fixed.

   {{sec-UCL}} defines a YANG module for policy-based network access control,
   which extends the IETF Access Control Lists (ACLs) module defined in {{!RFC8519}}.

>    This module can be used to ensure consistent enforcement of ACL policies

Reviewer: ```suggestion
   This  policy-based ACL module can be used to ensure consistent enforcement of ACL policies
```

   based on the group identity.

>    The ACL notion has been generalized to be device-nonspecific, and can be

Reviewer: ```suggestion
   The ACL concept has been generalized to be device-nonspecific, and can be
```

>    defined at network/administrative domain level {{?I-D.dbb-netmod-acl}}. To

Reviewer: ```suggestion
   defined at network/administrative domain level {{?I-D.ietf-netmod-acl-extensions}}. To
```

   allow for all applications of ACLs, the YANG module for policy-based network

>    access control defined in {{sec-UCL}} does not limit how it can be used.

Reviewer: ```suggestion
   ACL defined in {{sec-UCL}} does not limit how it can be used.
```

>    This document also defines a mechanism to establish a mapping between the

Reviewer: ```suggestion
   This document also defines a mechanism to establish a mapping between (1) the
```

>    user group identifier (ID) and common IP packet header fields and other

Reviewer: ```suggestion
   user group identifier (ID) and (2) common IP packet header fields and other
```

   encapsulating packet data (e.g., MAC address) to execute the policy-based access control.

   Additionally, the document defines a Remote Authentication Dial-in
   User Service (RADIUS) {{!RFC2865}} attribute that is used to
   communicate the user group identifier as part of identification and
   authorization information ({{sec-radius}}).

>    Although the document cites MAC address as an example in some sections, the

Reviewer: ```suggestion
   Although the document cites MAC addresses as an example in some sections, the
```

>    document does not make an assumption about which identifiers are used to trigger ACLs.

Reviewer: ```suggestion
   document does not make assumptions about which identifiers are used to trigger ACLs.
```

   The meanings of the symbols in tree diagrams are defined in
   {{?RFC8340}}.

   The document uses the following definitions and acronyms defined in {{!RFC8519}}:

>    * ACE

Reviewer: ```suggestion
   * Access Control Entry (ACE)   
```

>    * ACL

Reviewer: ```suggestion
   * Access Control List (ACL)
```

>    The follow term are used throughout this document:

Reviewer: ```suggestion
   The following terms are used throughout this document:
```

##  Overview {#overview}

   The architecture of a system that provides real-time and consistent

>    enforcement of access control policies is shown in {{arch}} and

Reviewer: ```suggestion
   enforcement of access control policies is shown in {{arch}}. This architecture

>    includes the following functional entities and interfaces.

Reviewer: ```suggestion
   includes the following functional entities and interfaces:
```

      which is responsible for enforcing appropriate access control
      policies.  In some cases, a PEP may map incoming packets to their
      associated source or destination endpoint-group IDs, and acts on

>       the endpoint-group ID based ACL policies, e.g., an NAS as the PEP

Reviewer: ```suggestion
      the endpoint-group ID based ACL policies, e.g., a NAS as the PEP
```

      Multiple PEPs may be involved in a network.

>       A PEP exposes a NETCONF interface to the SDN controller {{!RFC6241}}.

Reviewer: ```suggestion
      A PEP exposes a NETCONF interface {{!RFC6241}} to the SDN controller.
```

~~~~
{: #schedule-tree title="Schedule Tree Structure" artwork-align="center"}

>    The "period-of-time" allows a time period to be represented using either a start

Reviewer: ```suggestion
   The "period-of-time" allows a time period to be represented using either a start ("period-start")
```

>    and end date and time, or a start and a positive duration of time.

Reviewer: ```suggestion
   and end date and time ("period-end"), or a start and a positive time duration ("period-duration").
```

>    The "recurrence" allows an indication of the repeat scheduling of an event. The

Reviewer: ```suggestion
   The "recurrence" indicates the  scheduling recurrence of an event. The
```

   repetition can be scoped by a specified end time or by a count of occurences,
   and the frequency identifies the type of recurrence rule. For example, a "daily"
   frequency value specifies repeating events based on an interval of a day or more.
   The interval represents at which intervals the recurrence rule repeats. For example,
   within a daily recurrence rule, an interval value of "8" means every eight days.

>    An array of the bysecond/byminute/byhour specifies a list of seconds within a minute/

Reviewer: ```suggestion
   An array of the bysecond (or byminut, byhour) specifies a list of seconds within a minute (or
```

>    minutes within an hour/hours of the day.

Reviewer: ```suggestion
   minutes within an hour, hours of the day).
```

>    The parameter byday specifies a list of days of

Reviewer: ```suggestion
   The parameter "byday" specifies a list of days of
```

>    the week, with a optional direction indicates the nth orrurance of a specific day within

Reviewer: ```suggestion
   the week, with an optional direction which indicates the nth occurrence of a specific day within
```

   the "monthly" or "yearly" frequency. For example, within a "monthly" rule,

>    the weekday with a value of "monday" and the direction with a value of "-1"

Reviewer: ```suggestion
   the "weekday" with a value of "monday" and the "direction" with a value of "-1"
```

   represents the last Monday of the month.

>    An array of the bymonthday/byyearday/byyearweek/byyearmonth specifies a list of

Reviewer: ```suggestion
   An array of the "bymonthday" (or byyearday", "byyearweek", or "byyearmonth") specifies a list of
```

>    days of the month/days of the year/weeks of the year/months of the year.

Reviewer: ```suggestion
   days of the month (or days of the year, weeks of the year, or months of the year).
```

>    The bysetpos allows a list of values that corresponds to the nth occurance

Reviewer: ```suggestion
   The "bysetpos" conveys a list of values that corresponds to the nth occurrence
```

   within the set of recurrence instances to be specified. For example, in a "monthly"

>    recurrence rule, the byday parameter specifies every Monday of the week, the

Reviewer: ```suggestion
   recurrence rule, the "byday" data node specifies every Monday of the week, the
```

>    bysetpos with value of "-1" represents the last Monday of the month.

Reviewer: ```suggestion
   "bysetpos" with value of "-1" represents the last Monday of the month.
```

>    Not setting the bysetpos parameter represents every Monday of the month.

Reviewer: ```suggestion
   Not setting the "bysetpos" data node represents every Monday of the month.
```

>    The wkst parameter allows to specify the day on which the week starts. This is

Reviewer: ```suggestion
   The "wkst" data node specifies the day on which the week starts. This is
```

   significant when a "weekly" recurrence rule has an interval greater than 1, and

>    a byday parameter is specified. This is also significant when in a "yearly" rule

Reviewer: ```suggestion
   a "byday" data node is specified. This is also significant when in a "yearly" rule
```

>    and a byyearweek is specified. The default value is "monday".

Reviewer: ```suggestion
   and a "byyearweek" is specified. The default value is "monday".
```

~~~~
{
  "period-of-time": {
    "period-start": "2023-01-01T08:00:00Z",

>     "explicit-end": "2025-12-01T18:00:00Z"

Reviewer: ```suggestion
    "period-end": "2025-12-01T18:00:00Z"
```

~~~~
{
  "period-of-time": {
    "period-start": "2023-01-01T08:00:00Z",

>     "duration": "P15DT05:20:00"

Reviewer: ```suggestion
    "period-duration": "P15DT05:20:00"
```

~~~~
{
  "period-of-time": {
    "period-start": "2023-01-01T08:00:00Z",

>     "duration": "P20W"

Reviewer: ```suggestion
    "period-duration": "P20W"
```

~~~~

>    The following indicates the example of a recurrence that occurs every other week on Tuesday and Sunday, the week starts from Monday:

Reviewer: ```suggestion
   The following depicts the example of a recurrence that occurs every other week on Tuesday and Sunday, the week starts from Monday:
```

~~~

##  The UCL Extension to the ACL Model

>    This module specifies an extension to the IETF ACL model {{!RFC8519}} to add

Reviewer: ```suggestion
   This module specifies an extension to the IETF ACL model {{!RFC8519}}. This extension adds
```

~~~~
{: #ucl-tree title="UCL Extension" artwork-align="center"}

   The first part of the data model augments the "acl" list in the
   IETF ACL model {{!RFC8519}} with a "endpoint-groups" container

>    having a list of "endpoint group" inside, each entry has a group-id that uniquely

Reviewer: ```suggestion
   having a list of "endpoint group" inside, each entry has a "group-id" that uniquely
```

>    identifies the endpoint group. The choice statement allows for the selection

Reviewer: ```suggestion
   identifies the endpoint group. The choice statement controls the selection
```

   of group type between "user-group" or "device-group".

   The second part of the data model augments the "matches" container in the IETF
   ACL model {{!RFC8519}} so that a source and/or destination endpoint group index
   can be referenced as the macth criteria.

   The third part of the data model augments the "ace" list in the IETF ACL
   model {{!RFC8519}} with date and time specific parameters to allow ACE to be
   activated based on a date/time condition. Two types of time range are defined,

>    which reuses "recurrence" and "period" groupings defined in the "ietf-schedule"

Reviewer: ```suggestion
   which reuse "recurrence" and "period" groupings defined in the "ietf-schedule"
```

>    YANG model, respectively. Note that the data model augments the definition of

Reviewer: ```suggestion
   YANG module, respectively. Note that the data model augments the definition of
```

   "recurrence" grouping with a "duration" data node to specify the duration of

>    time for each occurance the policy activation is triggered.

Reviewer: ```suggestion
   time for each occurrence the policy activation is triggered.
```

   the group-based ACL.

   The PEP which enforces group-based ACL may acquire group identities

>    from the AAA server if working as an NAS authenticating both the

Reviewer: ```suggestion
   from the AAA server if working as a NAS authenticating both the
```
